src/dftbp/dftb/shortgammafuncs.F90

```diff
@@ -331,4 +331,593 @@ function gammaSubExprnPrime_(rab, tau1, tau2) result(gammaSubPrime)
+
+    if (rab < tolSameDist) then
+      write(errorString, "('Atoms on top of each other in gammaSubgExprnQuintuplePrime')")
+      call error(errorString)
+    end if
+
+    gammaSubgExprnQuintuplePrime = -120.0_dp / rab**6
+
+  end function gammaSubgExprnQuintuplePrime
+
+
+  !> Determines the value of the second derivative of the short range contribution to gamma with the
+  !> exponential form
+  function expGammaDoublePrime(rab, Ua, Ub)
+
+    !> separation of sites a and b
+    real(dp), intent(in) :: rab
+
+    !> Hubbard U for site a
+    real(dp), intent(in) :: Ua
+
+    !> Hubbard U for site b
+    real(dp), intent(in) :: Ub
+
+    !> returned contribution
+    real(dp) :: expGammaDoublePrime
+
+    real(dp) :: tauA, tauB, tauMean
+
+    if (rab < 0.0_dp) then
+      write(errorString, "('Failure in short-range gamma, r_ab negative :',f12.6)") rab
+      call error(errorString)
+    else if (Ua < MinHubTol) then
+      write(errorString, "('Failure in short-range gamma, U too small :',f12.6)") Ua
+      call error(errorString)
+    else if (Ub < MinHubTol) then
+      write(errorString, "('Failure in short-range gamma, U too small : ',f12.6)") Ub
+      call error(errorString)
+    end if
+
+    ! on-site case with R~0
+    if (rab < tolSameDist) then
+      if (abs(Ua - Ub) < MinHubDiff) then
+        ! same Hubbard U values, onsite , NOTE SIGN CHANGE!
+        tauMean = 3.2_dp * 0.5_dp * (Ua + Ub)
+        expGammaDoublePrime = -tauMean**3 / 48.0_dp
+      else
+        ! Ua /= Ub Hubbard U values - limiting case, NOTE SIGN CHANGE!
+        expGammaDoublePrime = 0.0_dp
+      end if
+    else if (abs(Ua - Ub) < MinHubDiff) then
+      ! R > 0 and same Hubbard U values
+      ! 16/5 * U, see review papers
+      tauMean = 3.2_dp * 0.5_dp * (Ua + Ub)
+      expGammaDoublePrime = &
+          & exp(-tauMean * rab) * gammaSubgExprnDoublePrime(rab, tauMean) &
+          & - 2.0_dp * tauMean * exp(-tauMean * rab) * gammaSubgExprnPrime(rab, tauMean) &
+          & + tauMean**2 * exp(-tauMean * rab) * gammaSubgExprn(rab, tauMean)
+    else
+      ! 16/5 * U, see review papers
+      tauA = 3.2_dp*Ua
+      tauB = 3.2_dp*Ub
+      ! using the sign convention in the review articles, not Joachim Elstner's thesis -- there's a
+      ! typo there
+      expGammaDoublePrime = &
+          & exp(-tauA * rab) * gammaSubfExprnDoublePrime(rab, tauA, tauB) &
+          & - 2.0_dp * tauA * exp(-tauA * rab) * gammaSubfExprnPrime(rab, tauA, tauB) &
+          & + tauA**2 * exp(-tauA * rab) * gammaSubfExprn(rab, tauA, tauB) &
+          & + exp(-tauB * rab) * gammaSubfExprnDoublePrime(rab, tauB, tauA) &
+          & - 2.0_dp * tauB * exp(-tauB * rab) * gammaSubfExprnPrime(rab, tauB, tauA) &
+          & + tauB**2 * exp(-tauB * rab) * gammaSubfExprn(rab, tauB, tauA)
+    end if
+  end function expGammaDoublePrime
+
+
+  !> Determines the value of the third derivative of the short range contribution to gamma with the
+  !> exponential form
```

Comment:
```suggestion
  !! exponential form
```
Only the first line of a multi-line comments should use !> acording to Doxygen mark-up